src/zarr/testing/store.py

```diff
@@ -103,6 +104,18 @@ async def test_set(self, store: S, key: str, data: bytes) -> None:
         observed = self.get(store, key)
         assert_bytes_equal(observed, data_buf)
 
+    async def test_set_dict(self, store: S) -> None:
+        """
+        Test that a dict of key : value pairs can be inserted into the store via the
+        `_set_dict` method.
+        """
+        keys = ["zarr.json", "c/0", "foo/c/0.0", "foo/0/0"]
+        data_buf = [Buffer.from_bytes(k.encode()) for k in keys]
+        store_dict = dict(zip(keys, data_buf, strict=True))
+        await store._set_dict(store_dict)
+        for k, v in store_dict.items():
+            assert self.get(store, k).to_bytes() == v.to_bytes()
```

Comment:
if x.to_bytes() == y.to_bytes(), does x== y?

Isn't there a multiple get? Maybe not important here.

Reply:
> if x.to_bytes() == y.to_bytes(), does x== y?

no, and I _suspect_ this might be deliberate since in principle `Buffer` instances can have identical bytes but different devices (e.g., gpu memory vs host memory); thus `x == y` might only be true if two buffers are bytes-equal _and_ device-equal, but I'm speculating here. @madsbk would have a better answer I think.

> Isn't there a multiple get? Maybe not important here.

there is no multiple get (nor a multiple set, nor a multiple delete).

Reply:
xref: [src/zarr/buffer.py](https://github.com/zarr-developers/zarr-python/pull/2006/files#diff-3affa2119d5b734fc851307a6ddd275af09dc0650a97a95cc82fa7ae30272a27R258) in https://github.com/zarr-developers/zarr-python/pull/2006